walkthroughs/cdk-examples/app-mesh-hands-on/lib/constructs/appmesh-fargate-service.ts

```diff
@@ -0,0 +1,90 @@
+import * as ecs from "aws-cdk-lib/aws-ecs";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Construct } from "constructs";
+import { MeshStack } from "../stacks/mesh-components";
+import { AppMeshFargateServiceProps } from "../utils";
+
+export class AppMeshFargateService extends Construct {
+  taskDefinition: ecs.FargateTaskDefinition;
+  service: ecs.FargateService;
+  securityGroup: ec2.SecurityGroup;
+  envoySidecar: ecs.ContainerDefinition;
+  xrayContainer: ecs.ContainerDefinition;
+
+  appContainer: ecs.ContainerDefinition;
+
+  constructor(ms: MeshStack, id: string, props: AppMeshFargateServiceProps) {
+    super(ms, id);
+
+    this.securityGroup = new ec2.SecurityGroup(this, `${props.serviceName}TaskSecurityGroup`, {
+      vpc: ms.sd.base.vpc,
+    });
+    this.securityGroup.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(ms.sd.base.PORT));
+
+    const proxyConfiguration =
+      props.envoyConfiguration &&
+      props.envoyConfiguration.container &&
+      props.envoyConfiguration.proxyConfiguration
```

Comment:
Since the `proxyConfiguration` needs the `container` and not the other way around,  in the `envoyConfiguration` prop, adding a `container` prop is mandatory but the `proxyConfiguration` prop can remain undefined (which is the default value).


```
export interface EnvoyConfiguration {
  container?: EnvoySidecar;
  proxyConfiguration?: ecs.ProxyConfiguration;
}
```